File: concrete/src/test/java/pcb/uwu/amount/composite/electromagnetism/CoulombsTest.java
package pcb.uwu.amount.composite.electromagnetism;

import org.junit.Test;
import pcb.uwu.amount.composite.termodynamics.Joules;
import pcb.uwu.amount.quantity.Amperes;
import pcb.uwu.amount.quantity.Seconds;

import static java.math.MathContext.DECIMAL64;
import static org.junit.Assert.assertEquals;
import static pcb.uwu.core.Magnitude.MICRO;
import static pcb.uwu.core.Magnitude.MILLI;

public class CoulombsTest {
	private static final Coulombs coulombs = new Coulombs(50, MILLI);

	@Test
	public void dividedByTime() {
		assertEquals(new Amperes(2, MILLI),
				coulombs.dividedBy(new Seconds(25), DECIMAL64));
	}

	@Test
	public void dividedByElectricCurrent() {
		assertEquals(new Seconds(25),
				coulombs.dividedBy(new Amperes(2, MILLI), DECIMAL64));
	}

	@Test
	public void dividedByElectricCapacitance() {
		assertEquals(new Volts(125),
				coulombs.dividedBy(new Farads(400, MICRO), DECIMAL64));
	}

	@Test
	public void dividedByElectricPotential() {
		assertEquals(new Farads(400, MICRO),
				coulombs.dividedBy(new Volts(125), DECIMAL64));
	}

	@Test
	public void multipliedByElectricPotential() {
		assertEquals(new Joules(5),
				coulombs.multipliedBy(new Volts(100), DECIMAL64));
	}
}

File: concrete/src/test/java/pcb/uwu/amount/derived/electromagnetism/CoulombsTest.kt
package pcb.uwu.amount.derived.electromagnetism

import org.junit.Assert.assertEquals
import org.junit.Test
import pcb.uwu.amount.base.Amperes
import pcb.uwu.amount.base.Seconds
import pcb.uwu.core.Magnitude.MICRO
import pcb.uwu.core.Magnitude.MILLI
import java.math.MathContext.DECIMAL64

class CoulombsTest
{
    @Test
    fun dividedByTime() =
        assertEquals("0.002 A",
                     Coulombs(50, MILLI).dividedBy(Seconds(25), DECIMAL64).toString())

    @Test
    fun dividedByElectricCurrent() =
        assertEquals("25 s",
                     Coulombs(50, MILLI).dividedBy(Amperes(2, MILLI), DECIMAL64)
                         .toString())

    @Test
    fun dividedByElectricCapacitance() =
        assertEquals("125 V",
                     Coulombs(50, MILLI).dividedBy(Farads(400, MICRO), DECIMAL64)
                         .toString())

    @Test
    fun dividedByElectricPotential() =
        assertEquals("0.0004 F",
                     Coulombs(50, MILLI).dividedBy(Volts(125), DECIMAL64).toString())

    @Test
    fun multipliedByElectricPotential() =
        assertEquals("5 J",
                     Coulombs(50, MILLI).multipliedBy(Volts(100), DECIMAL64).toString())
}